src/Caching/Memory/test/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -121,6 +122,34 @@ public async Task GetOrCreateAsync_AddsNewValue()
             Assert.Same(obj, result);
         }
 
+        [Fact]
+        public async Task GetOrCreateAsync_DoesNotBlock()
```

Comment:
```suggestion
        public async Task GetOrCreateAsync_AvoidsCapturingSynchronizationContextWhenAwaitingUseCallback()
```

It certainly can *block* depending on what the user callback does.